app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionActivity.kt

```diff
@@ -1,5 +1,105 @@
 package com.bogdan.codeforceswatcher.features.actions
 
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.KeyEvent
+import android.view.KeyEvent.KEYCODE_BACK
+import android.view.View
+import android.webkit.WebSettings
+import android.webkit.WebView
+import android.webkit.WebViewClient
 import androidx.appcompat.app.AppCompatActivity
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.features.actions.models.CFAction
+import com.bogdan.codeforceswatcher.store
+import kotlinx.android.synthetic.main.activity_action.*
 
-class ActionActivity : AppCompatActivity()
\ No newline at end of file
+class ActionActivity : AppCompatActivity() {
+
+    private var cfAction: CFAction? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_action)
+
+        setSupportActionBar(toolbar)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setDisplayShowHomeEnabled(true)
+
+        val commentId = intent.getLongExtra(COMMENT_ID, -1)
+        cfAction = store.state.actions.actions.find { it.comment?.id == commentId }
```

Comment:
If it's used only in `initViews` you can make it local and pass directly if non-optional, otherwise just finish activity.